### ISSUE TITLE
(Idea) Alternative building process for Windows: Simpler for the end-user & has consistent results

### ISSUE THREAD

**Issue by snowbie** [ImageMagick/ImageMagick#5094]
### Is your feature request related to a problem? Please describe.

Building ImageMagick has always given me the hardest time. In fact, I, to this day, still haven't been able to successfully build it with the configuration I want. Using the Visual Studio method is simple, but codec and feature use is unusually limited (no fftw??)
So this made me think about what could be done to fix this. As Windows users are certainly the minority when it comes to ImageMagick, I began thinking that the building process should be more like Linux's. Then the idea hit me, and I think it should be the new standard as it should be significantly better than the current method for Windows, for users and the devs.

### Describe the solution you'd like

A program that's often suggested for building ffmpeg is [media-autobuild_suite](https://github.com/m-ab-s/media-autobuild_suite), which allows you to select everything you want for your own custom build. It can also build MPV. What makes this special to me is that I can have these programs built with universal codec support. I connected the dots between the two, and I believe that the ImageMagick building process could be replicated using a fork of that. 

It's very simple for the end-user; you run the batch file and you pick the codecs (delegates) you want. Once that's done, all the user has to do is sit back and watch as the program is built.

### Additional context

-------
For the time being, I would really appreciate if someone could build ImageMagick for me. I currently use a ten year old, 32bit version only because it has fftw. I would really like to have a 64bit executable and ideally with as many codecs as possible. Below are the parameters I tried building.
```cmd
--enable-shared=no --enable-static=yes --with-quantum-depth=16 --enable-hdri=yes --enable-hugepages=yes --enable-legacy-support=yes --enable-gcov=yes --enable-pipes=yes --enable-gprof=yes --enable-prof=yes --with-tcmalloc=yes --enable-delegate-build --with-autotrace=yes --with-bzlib=yes --with-djvu=yes --with-dps=yes --enable-cipher=yes --with-fftw=yes --with-flif=yes --with-fontconfig=yes --with-fpx=yes --with-freetype=yes --with-gslib=yes --with-gvc=yes --with-heic=yes --with-jbig=yes --with-jpeg=yes --with-jxl=yes --with-lcms=yes --with-lqr=yes --with-magick-plus-plus=yes --with-openexr=yes --with-openjp2=yes --with-pango=yes --with-perl=yes --with-png=yes --with-raqm=yes --with-raw=yes --with-rsvg=yes --with-tiff=yes --with-webp=yes --with-wmf=yes --with-x=yes --with-xml=yes --with-zlib=yes --with-zstd=yes --with-windows-font-dir=C:/Windows/Fonts --with-dejavu-font-dir=C:/Windows/Fonts
```
I would give making the **magick-autobuild_suite** a shot, but it is currently finals week. I would like to hear other users' input.  
Does this sound like a good idea? Do you think it'd be superior to the existing methods?

**Comment by dlemstra** [comment]
I don't think we should invest time in this. We might switch to `cmake` in the future but for now the Windows build uses configure. And most codecs/features are supported on Windows.

The reason that there is no fftw support is because that is an incompatible license (GPL) and that cannot be used together with our apache license. And because we want our users to be able to use this commercially without any issues we cannot include support for fftw in our installers. And we did not invest time in making it possible to use this on Windows.

But I have good news I just tried to see how easy it would be to add fftw support and that turned out to be quite easy. To use this you will need to enable the "Build projects that are not compatible with the Apache 2.0 license" option when running `configure.exe` to create the solution and projects.

**Comment by snowbie** [comment]
> But I have good news I just tried to see how easy it would be to add fftw support and that turned out to be quite easy. To use this you will need to enable the "Build projects that are not compatible with the Apache 2.0 license" option when running `configure.exe` to create the solution and projects.

Thank you so much for doing that. It took some time to build, but in the end I was eventually able to get it all :) 
I'm assuming this is due to the licensing issue, but building with fftw required going to[ fftw's website](http://fftw.org/install/windows.html) and manually downloading the precompiled binaries and creating the library for IM to use.
